### PR TITLE
[ggj] feat: boxed/primitive type equality for if statement

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
[4/6] support boxed/primitive type equality in if statement.

More details can be found in notes: go/javanaut-boxed-primitive-type-equality

### CODE REVIEW

File: src/main/java/com/google/api/generator/engine/ast/IfStatement.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public IfStatement build() {
       IfStatement ifStatement = autoBuild();
       Preconditions.checkState(
           ifStatement.conditionExpr().type().equals(TypeNode.BOOLEAN),
-          "If-condition must be a boolean-typed expression");
+          "If-condition must be a boolean/Boolean-typed expression");
```
Comment (miraleung) [inline src/main/java/com/google/api/generator/engine/ast/IfStatement.java:68]: No need to update the comment, the existing one is clear enough to readers.

File: src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java [hunk 2]
```diff
@@ -1004,6 +1004,22 @@ public void writeIfStatement_simple() {
         String.format("%s%s%s%s", "if (condition) {\n", "int x = 3;\n", "int x = 3;\n", "}\n"));
   }
 
+  @Test
+  public void writeIfStatement_booleanObjectCondition() {
```
Comment (miraleung) [inline src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java:1008]: This test should go in IfStatementTest, could we please move it?

File: src/main/java/com/google/api/generator/engine/ast/IfStatement.java (hunk 1 above)
Comment (xiaozhenliu-gg5) [reply src/main/java/com/google/api/generator/engine/ast/IfStatement.java]: Done.

File: src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java (hunk 2 above)
Comment (xiaozhenliu-gg5) [reply src/test/java/com/google/api/generator/engine/writer/JavaWriterVisitorTest.java]: Sure, added `IfStatementTest` and moved the booleanObjectCondition there.

File: src/test/java/com/google/api/generator/engine/ast/IfStatementTest.java [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            .setConditionExpr(nestedCondExpr)
+            .setBody(createAssignmentExprList())
+            .setElseBody(createAssignmentExprList())
+            .build();
+    IfStatement ifStatement =
+        IfStatement.builder()
+            .setConditionExpr(condExpr)
+            .setBody(Arrays.asList(nestedIfStatement))
+            .build();
+    assertTrue(ifStatement.conditionExpr().type().equals(TypeNode.BOOLEAN));
```
Comment (miraleung) [inline src/test/java/com/google/api/generator/engine/ast/IfStatementTest.java:81]: `assertEquals`? Same elsewhere.

Comment (xiaozhenliu-gg5) [reply src/test/java/com/google/api/generator/engine/ast/IfStatementTest.java:81]: Fixed.

**Pull request merged**